primefaces/src/main/java/org/primefaces/component/tristatecheckbox/TriStateCheckboxRenderer.java

```diff
@@ -119,11 +120,11 @@ protected void encodeInput(final FacesContext context, final TriStateCheckbox ch
     }
 
     protected void encodeOutput(final FacesContext context, final TriStateCheckbox checkbox, final int valCheck,
-                                final boolean disabled) throws IOException {
+                                final boolean disabled, boolean readonly) throws IOException {
         ResponseWriter writer = context.getResponseWriter();
         String styleClass = createStyleClass(checkbox, null, HTML.CHECKBOX_BOX_CLASS) ;
         styleClass = (valCheck == 1 || valCheck == 2) ? styleClass + " ui-state-active" : styleClass;
-        styleClass = disabled ? styleClass + " ui-state-disabled" : styleClass;
```

Comment:
I've not tested and I didn't investigate, but are you sure this line should be deleted?

Reply:
No this is a mistake!